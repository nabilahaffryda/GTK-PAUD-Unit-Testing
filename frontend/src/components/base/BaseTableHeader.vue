<template>
  <v-card flat width="100%">
    <v-card-text :class="!searchInput ? 'py-1' : ''">
      <v-row no-gutters>
        <v-col cols="12" md="3" class="my-auto">
          <v-row no-gutters>
            <v-col>
              <v-list-item class="pa-0">
                <v-list-item-content class="pa-0">
                  <v-list-item-title class="title">
                    <slot name="title" />
                  </v-list-item-title>
                  <v-list-item-subtitle>
                    <slot name="subtitle" />
                  </v-list-item-subtitle>
                </v-list-item-content>
              </v-list-item>
            </v-col>
            <v-col class="hidden-md-and-up">
              <v-toolbar flat dense>
                <slot name="toolbar" />
                <v-spacer></v-spacer>
                <v-btn small icon @click="add" v-if="btnAdd">
                  <v-icon color="secondary">mdi-plus</v-icon>
                </v-btn>
                <v-btn small icon @click="download" v-if="btnDownload">
                  <v-icon color="secondary">mdi-download</v-icon>
                </v-btn>
                <v-btn small icon @click="filter" v-if="btnFilter">
                  <v-icon color="secondary">mdi-filter-variant</v-icon>
                </v-btn>
                <v-btn small icon @click="reload" v-if="btnReload">
                  <v-icon color="secondary">mdi-reload</v-icon>
                </v-btn>
              </v-toolbar>
            </v-col>
          </v-row>
        </v-col>
        <v-col cols="12" md="9">
          <div class="hidden-md-and-up">
            <v-text-field
              v-if="searchInput"
              v-model="keyword"
              :label="searchLabel"
              dense
              single-line
              outlined
              hide-details
              class="mt-1 mr-1"
              append-icon="mdi-magnify"
              @keyup.enter="search"
              @click:append="search"
            >
            </v-text-field>
          </div>
          <div class="hidden-sm-only hidden-xs-only">
            <v-toolbar flat>
              <v-spacer></v-spacer>
              <v-text-field
                v-if="searchInput"
                v-model="keyword"
                :label="searchLabel"
                dense
                single-line
                outlined
                hide-details
                class="my-auto mr-1"
                append-icon="mdi-magnify"
                @keyup.enter="search"
                @click:append="search"
              >
              </v-text-field>
              <div style="min-width: 8%">
                <v-tooltip top>
                  <template v-slot:activator="{ on, attrs }">
                    <v-btn
                      class="button-action"
                      outlined
                      depressed
                      small
                      width="40px"
                      height="40px"
                      v-bind="attrs"
                      v-on="on"
                      @click="upload"
                      v-if="btnUpload"
                    >
                      <v-icon color="primary">mdi-upload</v-icon>
                    </v-btn>
                  </template>
                  <span>Unggah</span>
                </v-tooltip>
                <v-tooltip top>
                  <template v-slot:activator="{ on, attrs }">
                    <v-btn
                      class="button-action"
                      outlined
                      depressed
                      small
                      width="40px"
                      height="40px"
                      v-bind="attrs"
                      v-on="on"
                      @click="filter"
                      v-if="btnFilter"
                    >
                      <v-icon color="primary">mdi-filter-variant</v-icon>
                    </v-btn>
                  </template>
                  <span>Saring</span>
                </v-tooltip>
                <v-tooltip top>
                  <template v-slot:activator="{ on, attrs }">
                    <v-btn
                      class="button-action"
                      outlined
                      depressed
                      small
                      width="40px"
                      height="40px"
                      v-bind="attrs"
                      v-on="on"
                      @click="download"
                      v-if="btnDownload"
                    >
                      <v-icon color="primary">mdi-download</v-icon>
                    </v-btn>
                  </template>
                  <span>Unduh</span>
                </v-tooltip>
                <v-tooltip top>
                  <template v-slot:activator="{ on, attrs }">
                    <v-btn
                      class="button-action"
                      outlined
                      depressed
                      small
                      width="40px"
                      height="40px"
                      v-bind="attrs"
                      v-on="on"
                      @click="reload"
                      v-if="btnReload"
                    >
                      <v-icon color="primary">mdi-reload</v-icon>
                    </v-btn>
                  </template>
                  <span>Muat Ulang</span>
                </v-tooltip>
                <v-tooltip top>
                  <template v-slot:activator="{ on, attrs }">
                    <v-btn
                      color="primary"
                      depressed
                      small
                      width="40px"
                      height="40px"
                      v-bind="attrs"
                      v-on="on"
                      @click="add"
                      v-if="btnAdd"
                    >
                      <v-icon color="white">mdi-plus</v-icon>
                    </v-btn>
                  </template>
                  <span>Tambah</span>
                </v-tooltip>
              </div>
              <slot name="toolbar" style="width: 89%" />
            </v-toolbar>
          </div>
        </v-col>
      </v-row>
    </v-card-text>
  </v-card>
</template>

<script>
export default {
  name: 'BaseTableHeader',
  props: {
    options: {
      type: Object,
      default: () => {},
    },
    searchInput: {
      type: Boolean,
      default: true,
    },
    searchLabel: {
      type: String,
      default: 'Cari Data',
    },
    btnFilter: {
      type: Boolean,
      default: false,
    },
    btnReload: {
      type: Boolean,
      default: true,
    },
    btnAdd: {
      type: Boolean,
      default: false,
    },
    btnUpload: {
      type: Boolean,
      default: false,
    },
    btnDownload: {
      type: Boolean,
      default: false,
    },
    classToolbar: {
      type: String,
      default: 'mb-6',
    },
  },
  data() {
    return {
      keyword: '',
    };
  },
  methods: {
    search() {
      this.$emit('search', this.keyword);
    },

    filter() {
      this.$emit('filter');
    },

    reload() {
      this.$emit('reload');
    },

    add() {
      this.$emit('add');
    },

    download() {
      this.$emit('download');
    },

    upload() {
      this.$emit('upload');
    },

    mutate(value) {
      this.$emit('keyword', value);
    },
  },

  watch: {
    keyword: 'mutate',
  },
};
</script>
<style scoped>
.border-bottom {
  border-bottom: 1px solid #a3a3a3;
}
.button-action {
  margin-right: 3px;
  border: solid 1px grey;
}
</style>
