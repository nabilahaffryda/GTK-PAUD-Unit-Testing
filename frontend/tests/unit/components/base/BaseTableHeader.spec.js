import { mount, createLocalVue } from "@vue/test-utils";
import BaseTableHeader from "@/components/base/BaseTableHeader.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseTableHeader.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('render table header', () => {
        const VCard = {
            props: ['options', 'searchInput', 'searchLabel', 'btnFilter', 'btnReload', 'btnAdd', 'btnUpload', 'btnDownload', 'classToolbar'],
            template: '<div><slot :label="searchLabel" /></div>'
        }

        const wrapper = mount(BaseTableHeader, {
            localVue,
            vuetify,
            stubs: {
                VCard
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})
