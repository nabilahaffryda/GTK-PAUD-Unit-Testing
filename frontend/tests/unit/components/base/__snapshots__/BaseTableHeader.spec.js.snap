// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTableHeader.vue render table header 1`] = `
<div flat="" width="100%">
  <div class="v-card__text">
    <div class="row no-gutters">
      <div class="my-auto col-md-3 col-12">
        <div class="row no-gutters">
          <div class="col">
            <div tabindex="-1" class="pa-0 v-list-item theme--light">
              <div class="v-list-item__content pa-0">
                <div class="v-list-item__title title"></div>
                <div class="v-list-item__subtitle"></div>
              </div>
            </div>
          </div>
          <div class="hidden-md-and-up col">
            <header class="v-sheet theme--light v-toolbar v-toolbar--dense v-toolbar--flat" style="height: 48px;">
              <div class="v-toolbar__content" style="height: 48px;">
                <div class="spacer"></div>
                <!---->
                <!---->
                <!----> <button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--small"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-reload theme--light secondary--text"></i></span></button>
              </div>
            </header>
          </div>
        </div>
      </div>
      <div class="col-md-9 col-12">
        <div class="hidden-md-and-up">
          <div class="v-input mt-1 mr-1 v-input--hide-details v-input--dense theme--light v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outlined">
            <div class="v-input__control">
              <div class="v-input__slot">
                <fieldset aria-hidden="true">
                  <legend><span class="notranslate">​</span></legend>
                </fieldset>
                <div class="v-text-field__slot"><label for="input-7" class="v-label theme--light" style="left: 0px; position: absolute;">Cari Data</label><input id="input-7" type="text"></div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><button type="button" aria-label="append icon" class="v-icon notranslate v-icon--link mdi mdi-magnify theme--light"></button></div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="hidden-sm-only hidden-xs-only">
          <header class="v-sheet theme--light v-toolbar v-toolbar--flat" style="height: 64px;">
            <div class="v-toolbar__content" style="height: 64px;">
              <div class="spacer"></div>
              <div class="v-input my-auto mr-1 v-input--hide-details v-input--dense theme--light v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outlined">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <fieldset aria-hidden="true">
                      <legend><span class="notranslate">​</span></legend>
                    </fieldset>
                    <div class="v-text-field__slot"><label for="input-10" class="v-label theme--light" style="left: 0px; position: absolute;">Cari Data</label><input id="input-10" type="text"></div>
                    <div class="v-input__append-inner">
                      <div class="v-input__icon v-input__icon--append"><button type="button" aria-label="append icon" class="v-icon notranslate v-icon--link mdi mdi-magnify theme--light"></button></div>
                    </div>
                  </div>
                </div>
              </div>
              <div style="min-width: 8%;"><span class="v-tooltip v-tooltip--top"><!----></span> <span class="v-tooltip v-tooltip--top"><!----></span> <span class="v-tooltip v-tooltip--top"><!----></span> <span class="v-tooltip v-tooltip--top"><!----><button type="button" class="button-action v-btn v-btn--outlined theme--light v-size--small" style="height: 40px; width: 40px;" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-reload theme--light primary--text"></i></span></button></span> <span class="v-tooltip v-tooltip--top"><!----></span></div>
            </div>
          </header>
        </div>
      </div>
    </div>
  </div>
</div>
`;
